color/src/gradient.rs

```diff
@@ -181,3 +184,158 @@ impl<CS: ColorSpace> Iterator for GradientIter<CS> {
         }
     }
 }
+
+/// The iterator for gradient approximation.
+///
+/// This will yield a value for each gradient stop, including `t` values
+/// of 0 and 1 at the endpoints.
+///
+/// Use the [`gradient_unpremultiplied`] function to generate this iterator.
+///
+/// Similar to [`GradientIter`], but does interpolation in unpremultiplied (straight) alpha space
+/// as specified in [HTML 2D Canvas].
+///
+/// [HTML 2D Canvas]: https://html.spec.whatwg.org/multipage/#interpolation
+#[expect(missing_debug_implementations, reason = "it's an iterator")]
+pub struct UnpremultipliedGradientIter<CS: ColorSpace> {
+    interpolator: UnpremultipliedInterpolator,
+    // This is in deltaEOK units
+    tolerance: f32,
+    // The adaptive subdivision logic is lifted from the stroke expansion paper.
+    t0: u32,
+    dt: f32,
+    target0: AlphaColor<CS>,
+    target1: AlphaColor<CS>,
+    end_color: AlphaColor<CS>,
+}
+
+/// Generate a piecewise linear approximation to a gradient ramp without alpha premultiplication.
+///
+/// Similar to [`gradient`], but does interpolation in unpremultiplied (straight) alpha space
+/// as specified in [HTML 2D Canvas]
+/// Similar to [`gradient`], but colors are interpolated without premultiplying their color
+/// channels by the alpha channel. This is almost never what you want.
+///
+/// This causes color information to leak out of transparent colors. For example, when
+/// interpolating from a fully transparent red to a fully opaque blue in sRGB, this
+/// method will go through an intermediate purple.
+///
+/// This matches behavior of gradients in the HTML `canvas` element.
+/// See [The 2D rendering context § Fill and stroke styles][HTML 2D Canvas] of the
+/// HTML 2D Canvas specification.
+/// [HTML 2D Canvas]: <https://html.spec.whatwg.org/multipage/#interpolation>
```

Comment:
A newline is required to get the link to render here.

```suggestion
/// This matches behavior of gradients in the HTML `canvas` element.
/// See [The 2D rendering context § Fill and stroke styles][HTML 2D Canvas] of the
/// HTML 2D Canvas specification.
///
/// [HTML 2D Canvas]: <https://html.spec.whatwg.org/multipage/#interpolation>
```